compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirArrayOfCallTransformer.kt

```diff
@@ -51,4 +54,19 @@ internal class FirArrayOfCallTransformer {
             replaceTypeRef(typeRef)
         }
     }
+
+    companion object {
+        private val arrayOfNames = listOf("emptyArray", "arrayOf") +
+                listOf(
+                    "boolean", "byte", "char", "double", "float", "int", "long", "short",
+                    "ubyte", "uint", "ulong", "ushort"
+                ).map { it + "ArrayOf" }
+
+        private fun arrayOfCallParameterCheckers(name: Name): Function2<List<FirValueParameter>, List<FirExpression>, Boolean> =
```

Comment:
Seems unnecessary to create a new lambda instance every time. This could just be:

```kt
private fun isArrayOf(function: FirFunction, arguments: List<FirExpression>): Boolean =
  when (function.name.asString()) {
    "emptyArray" -> function.valueParameters.isEmpty() && arguments.isEmpty()
    in arrayOfNames -> function.valueParameters.size == 1 && function.valueParameters[0].isVararg && arguments.size == 1
    else -> false
  }
```

...or you could just fold this all into `isArrayOfCall` and combine the other conditions.

Reply:
Thank you for the suggestion! Changed.